rasa_sdk/endpoint.py

```diff
@@ -90,7 +92,13 @@ async def health(_) -> HTTPResponse:
     @app.post("/webhook")
     async def webhook(request: Request) -> HTTPResponse:
         """Webhook to retrieve action calls."""
-        action_call = request.json
+        if request.headers.get("Content-Encoding") == "deflate":
+            # Decompress the request data using zlib
+            decompressed_data = zlib.decompress(request.body)
+            # Load the JSON data from the decompressed request data
+            action_call = json.loads(decompressed_data)
```

Comment:
Are there any exceptions that could be thrown during decompression that we need to handle?

Reply:
Not really. This is non-recoverable error per-request.

Reply:
but that's a good point. I think both decompression errors + json decoding error be caught, logged and a 400 response should be returned (similar to line 104 below)